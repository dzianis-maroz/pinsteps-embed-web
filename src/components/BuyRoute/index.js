
export const BuyRoute = (props) => {

    const { id } = props.match.params;

    const routeUrl = `https://pinsteps.page.link/?amv=35&apn=com.pinsteps&efr=0&ibi=com.pinsteps.ios&isi=1271374440&link=https%3A%2F%2Fweb.pinsteps.com%2Fshare%2F${id}%3Fl%3Den`;

    return (
        <div>
            <h1>Buy Route</h1>
            <a href = {routeUrl}><button>Get route</button></a>
        </div>
    )
}; 

export default BuyRoute;