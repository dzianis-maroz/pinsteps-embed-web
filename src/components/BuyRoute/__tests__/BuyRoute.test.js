import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

import BuyRoute from "../";

configure({ adapter: new Adapter() });

describe("Should render correctly", () => {
  const props = {
        match: {
            params: {
                id: 140,
            },
        },
    };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<BuyRoute {...props} />);
  })

  it("to Match Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("component is defined", () => {
    console.log(wrapper.props())
    expect(wrapper).toHaveLength(1);
  });

});
