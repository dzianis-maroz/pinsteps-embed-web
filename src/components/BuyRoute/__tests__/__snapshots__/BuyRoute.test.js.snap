// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly to Match Snapshot 1`] = `
<div>
  <h1>
    Buy Route
  </h1>
  <a
    href="https://pinsteps.page.link/?amv=35&apn=com.pinsteps&efr=0&ibi=com.pinsteps.ios&isi=1271374440&link=https%3A%2F%2Fweb.pinsteps.com%2Fshare%2F140%3Fl%3Den"
  >
    <button>
      Get route
    </button>
  </a>
</div>
`;
